src/plugins/workspace/public/components/workspace_list/index.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiPageHeader,
+  EuiPageContent,
+  EuiBasicTable,
+  EuiLink,
+  Direction,
+  CriteriaWithPagination,
+} from '@elastic/eui';
+import useObservable from 'react-use/lib/useObservable';
+import { useMemo } from 'react';
+import { useCallback } from 'react';
+import { WorkspaceAttribute } from '../../../../../core/public';
+
+import { useOpenSearchDashboards } from '../../../../../plugins/opensearch_dashboards_react/public';
+import { switchWorkspace } from '../utils/workspace';
+
+export const WorkspaceList = () => {
+  const {
+    services: { workspaces, application },
+  } = useOpenSearchDashboards();
+
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(5);
+  const [sortField, setSortField] = useState<'name' | 'id'>('name');
+  const [sortDirection, setSortDirection] = useState<Direction>('asc');
+
+  const workspaceList = useObservable(workspaces!.client.workspaceList$, []);
+
+  const pageOfItems = useMemo(() => {
+    return workspaceList
+      .sort((a, b) => {
+        const compare = a[sortField].localeCompare(b[sortField]);
+        return sortDirection === 'asc' ? compare : -compare;
+      })
+      .slice(pageIndex * pageSize, (pageIndex + 1) * pageSize);
+  }, [workspaceList, pageIndex, pageSize, sortField, sortDirection]);
+
+  const handleSwitchWorkspace = useCallback(
+    (id: string) => {
+      if (workspaces && application) {
+        switchWorkspace({ workspaces, application }, id);
+      }
+    },
+    [workspaces, application]
+  );
+
+  const columns = [
+    {
+      field: 'name',
+      name: 'Name',
+      sortable: true,
+      render: (name: string, item: WorkspaceAttribute) => (
+        <span>
+          <EuiLink onClick={() => handleSwitchWorkspace(item.id)}>{name}</EuiLink>
```

Comment:
I wanted to keep the logic of switching workspace here consistent with the dropdown list if their logic is the same, so that it is controllable when the following update occurs.

Reply:
I am ok for both two implementations. Since we are using link here, just wondering if more easy to use href attribute directly here.

Reply:
OK, my original thought was to render the link ui through the Link component, and then unify the switch logic of dropdown list with here.